src/test/java/mate/academy/service/impl/RoleServiceImplTest.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.model.Role;
+import mate.academy.service.RoleService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class RoleServiceImplTest {
+    private static RoleService roleService;
+    private static RoleDao roleDao;
+    private static Role role;
+
+    @BeforeEach
+    void setUp() {
+        role = new Role();
+        role.setRoleName(Role.RoleName.USER);
+        roleDao = Mockito.mock(RoleDao.class);
+        roleService = new RoleServiceImpl(roleDao);
+        Mockito.when(roleDao.save(role)).thenReturn(role);
+        Mockito.when(roleDao.getRoleByName("USER"))
+                .thenReturn(Optional.ofNullable(role));
```

Comment:
it's better to configure mocks inside test methods since they may need different behavior